go/vt/mysqlctl/replication.go

```diff
@@ -114,6 +114,23 @@ func (mysqld *Mysqld) StopSlave(hookExtraEnv map[string]string) error {
 	return mysqld.executeSuperQueryListConn(ctx, conn, []string{conn.StopSlaveCommand()})
 }
 
+// StopSlaveIOThread stops a slave's IO thread only.
+func (mysqld *Mysqld) StopSlaveIOThread(hookExtraEnv map[string]string) error {
+	h := hook.NewSimpleHook("preflight_stop_slave_io_thread")
```

Comment:
@sougou what is our stance on hooks? Do we add a new one for every new action on MysqlDaemon?

Reply:
I want to deprecate them.

Reply:
I followed the pattern established by `StopSlave()` method on `Mysqld`. Should I remove the hook entirely @sougou?

Reply:
Go ahead and remove the new hook @PrismaPhonic. We can revisit if needed when you do the ERS PR.

Reply:
Done.